### PR TITLE
Calo score

### PR DESCRIPTION
**Pull request by tihsu99**
Update for the requirement dicussed in the meeting

### CODE REVIEW

File: util/evaluate_image_based.py [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
 from evaluate_plotting_helper import *
 
-torch.set_default_dtype(torch.float64)
+#torch.set_default_dtype(torch.float64)
```
Comment (Wilsker) [inline util/evaluate_image_based.py]: Is there a reason you've commented this line and then cast all the tensors individually later?

File: util/HighLevelFeatures.py [hunk 2]
```diff
@@ -129,6 +129,8 @@ def _DrawSingleLayer(self, data, layer_nr, filename, title=None, fig=None, subpl
         ax.grid(False)
         if vmax is None:
             vmax = data.max()
+            vmax = max(1e-1, vmax)
```
Comment (Wilsker) [inline util/HighLevelFeatures.py:134]: Can you print a warning message here to make sure we can debug plot issues easily please

File: util/evaluate_image_based.py (hunk 1 above)
Comment (tihsu99) [reply util/evaluate_image_based.py]: Yes, there is a issue related to this default dtype setting. Please see the discussion here. https://discuss.pytorch.org/t/tensors-of-the-same-index-must-be-on-the-same-device-and-the-same-dtype-except-step-tensors-that-can-be-cpu-and-float32-notwithstanding/190335/4

File: util/Evaluate.py [hunk 3]
```diff
@@ -63,7 +65,10 @@ def digitize_input(sample_list, particle, filename, dtype=torch.float32,pad_valu
 
         bin_indices = bin_indices.cpu().numpy().copy()
         E_          = output_[:,0].cpu().numpy().copy()
-        mask        = (~(E_ == pad_value)) & (E_ > 0.)
+        r_Mask      = r_Mask.cpu().numpy().copy()
+        theta_Mask  = theta_Mask.cpu().numpy().copy()
+        z_Mask      = z_Mask.cpu().numpy().copy()
+        mask        = (~(E_ == pad_value)) & (E_ > 0.015) & (~r_Mask) & (~theta_Mask) & (~z_Mask) #TODO: Check E unit, want to cut on 15keV
```
Comment (Wilsker) [inline util/Evaluate.py]: The TODO comment is already done right?

Comment (tihsu99) [reply util/Evaluate.py]: Done

File: util/evaluate_image_based.py (hunk 1 above)
Comment (Wilsker) [reply util/evaluate_image_based.py]: great, thanks, good to have this documented

File: util/HighLevelFeatures.py (hunk 2 above)
Comment (tihsu99) [reply util/HighLevelFeatures.py:134]: Done

File: util/evaluate_image_based.py (hunk 1 above)
Comment (tihsu99) [reply util/evaluate_image_based.py]: Comment edited to provide this link

File: util/Evaluate.py [hunk 4]
```diff
@@ -20,10 +20,12 @@
 def digitize(tensor, bin_edges, device, middle='true', dtype=torch.float32):
     bin_edges = torch.tensor(bin_edges, device=torch.device(device))
     bin_indices = torch.bucketize(tensor, bin_edges) - 1
+    Mask = (bin_indices >= len(bin_edges) - 1) | (bin_indices == 1)
```
Comment (Wilsker) [inline util/Evaluate.py:23]: Am I correct in my understanding that this is to mask out hits that fall into the last bin i.e. preventing it from displaying the 'overflow'?

**Review by Wilsker — APPROVED** [review]

File: util/Evaluate.py (hunk 4 above)
Comment (tihsu99) [reply util/Evaluate.py:23]: Yes, and also the first bin i.e. before the detector showering.
